android/hybridcommon-ui/src/main/java/com/revenuecat/purchases/hybridcommon/ui/PaywallHelpers.kt

```diff
@@ -31,18 +31,26 @@ fun presentPaywallFromFragment(
 }
 
 fun presentPaywallFromFragment(
-    fragment: FragmentActivity,
+    activity: FragmentActivity,
     options: PresentPaywallOptions,
 ) {
     with(options) {
-        fragment
+        val requestKey = System.identityHashCode(paywallResultListener).toString()
+
+        activity.supportFragmentManager.setFragmentResultListener(requestKey, activity) { _, result ->
```

Comment:
I'm wondering, does this survive activity recreation? Just making sure since I think that was the main reason to using a VM...